src/main.rs

```diff
@@ -117,7 +129,34 @@ impl Group {
         unimplemented!("WIP")
     }
 
-    async fn store_keypair(&self, protected_store: &veilid_core::ProtectedStore) -> Result<()> {
+    pub async fn close(&self) -> Result<()> {
+        let routing_context = &self.routing_context;
+        let key = self.dht_record.key().clone();
+        routing_context.close_dht_record(key).await?;
+        Ok(())
+    }
+
+    pub fn encrypt_aead(
+        &self,
+        data: &[u8],
+        associated_data: Option<&[u8]>,
+    ) -> Result<Vec<u8>> {
+        self.crypto_system
```

Comment:
The nonce needs to be generated for each run of encryption otherwise there's a risk of the secret key being leaked. We should store it along with the encrypted data since the nonce is public.

use the `crypto.random_nonce` function [here](https://gitlab.com/veilid/veilid/-/blob/main/veilid-core/src/crypto/vld0/mod.rs?ref_type=heads#L132)

The buffer we store should look like

```
[fixed bytes of nonce][encrypted blob]
```

Then we can split the buffer back into the nonce + encrypted data